### DISCUSSION TITLE
Disarm not working

### DISCUSSION THREAD

**Discussion by aidanbha79** [tsightler/ring-mqtt#703] [Support Q&A] [answered]
Hi, my ring alarm has been working with the integration for some time and not had an issue recently I am not able to disarm the system using HA. I have got rid of the subscription for ring. Could this be the issue?

**Comment by tsightler** [comment]
Possibly.  Ring announced that they would require subscriptions to access arm/disarm from the app if you purchased the system after March 29, 2023.  If you purchased before then you should be grandfathered in.

https://support.ring.com/hc/en-us/articles/12866034603028-An-Update-to-Ring-In-App-Features

The simple answer is, if you can do it with the app, you should be able to do it with ring-mqtt and, if not, then you can't.  If you can arm/disarm via the app but not ring-mqtt, then you should be sure to restart ring-mqtt and check logs for any issues.

**Reply by aidanbha79** [reply]
I have had it a long time so won't be that and I checked and the subscription was still active but still won't let me disarm it even through the keypad on the dashboard. It's very strange as it has been working fine.

**Reply by tsightler** [reply]
Did you restart the addon and check the logs?  There's no secrets, every single function is logged.  There was a global Ring outage about a week ago that caused many users to need to restart the addon to recover.  The addon tries to auto-recover from most issues, but there can be failures from which it doesn't know how to recover.  Many people think restarting HA restarts the addon, but it does not.

**Reply by aidanbha79** [reply]
I have restarted it today and still not working. I will have to check the logs and go through them. See if anything is coming up in them. It looks to be working but then it just doesn't trigger the actual action. The arm works a treat though.

**Reply by tsightler** [reply]
OK, well that is super weird if arming works but disarm doesn't.

**Reply by aidanbha79** [reply]
Any particular place or item I should look for in the logs? I can see the device status in the mqtt ring logs but can't really see anything in the main logs.

**Reply by tsightler** [reply]
When you perform the disarm you should then look at the logs to see if the disarm is getting to ring-mqtt or not.

**Reply by aidanbha79** [reply]
it says the wrong disarm code was entered but it really isn't. I have reset the code, restarted HA and tried again and nothing shows in the logs that it is even sending the request out to change the status.

**Reply by tsightler** [reply, marked as answer]
What do you mean by "reset the code"?  It kind of makes it sound like you did something in the Ring app.  You are aware that the code used for HA frontend is the one configured in the ring-mqtt config options, not the ones configured in the Ring app?

**Reply by aidanbha79** [reply]
I must have forgotten that part thank you for the help in getting it working again. I had changed the code in the app and it was pointing to the old one not the new one. Changed it in the config and all working now.

**Reply by tsightler** [reply]
Yeah, I really wish it was possible to use the codes from Ring but, so far, I haven't been able to work out a way to do this.  The Ring app doesn't use codes, and when you create codes they are simply pushed to the base station and the base station validates the codes locally when they are entered on the keypad.  There are some objects associated with the codes in the API, but they codes themselves don't appear to be accessible, perhaps due to the security risks.

**Reply by aidanbha79** [reply]
Yes I guess it would be a security issue if you could see the code. It just looks like you add the code in to the disarm screen on the HA app and that passes through.
